benchmark/utils/loggers.hpp

```diff
@@ -172,21 +172,15 @@ template <typename ValueType>
 struct ResidualLogger : gko::log::Logger {
     using rc_vtype = gko::remove_complex<ValueType>;
 
-    // TODO2.0: Remove when deprecating simple overload
-    void on_iteration_complete(const gko::LinOp* solver,
-                               const gko::size_type& it,
-                               const gko::LinOp* residual,
+    void on_iteration_complete(const gko::LinOp*,
```

Comment:
Does this compile without warnings? GCC warns about partially overridden functions, which is why I added the other override and the TODO

Reply:
No issues at least with gcc11 and our default settings for debug. Do we have an option to increase the warning level?

Reply:
Still nothing in that file, but so many other warnings. Perhaps we should add more warnings to our default debug build.